### PR TITLE
Update ember-cli from ember example

### PR DESCRIPTION
**Pull request by BarryThePenguin**
Issue: #13142

## What I did

I've taken the upgrade work I did in my previous PR and pulled it out into this PR

### CODE REVIEW

**Comment by shilman** [comment]
@BarryThePenguin we're getting the following error in CI. Can you look into it?

```
ERR! => Failed to build the preview
ERR! Module not found: Error: Can't resolve '../ember-output/storybook-docgen/index.json' in '/tmp/storybook/examples/ember-cli/.storybook'
ERR! ModuleNotFoundError: Module not found: Error: Can't resolve '../ember-output/storybook-docgen/index.json' in '/tmp/storybook/examples/ember-cli/.storybook'
```

**Comment by BarryThePenguin** [comment]
Sure! I think I known what it is.. there's a fix for it in my other PR. I'll take a look today 👀

**Comment by BarryThePenguin** [comment]
Looks like I inadvertently changed something in https://github.com/storybookjs/ember-cli-storybook/pull/50, which changed how addon options are configured, specifically this line https://github.com/storybookjs/ember-cli-storybook/pull/50/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346L11-R11

Previously it was expecting..

```js
'ember-cli-storybook': {
  enableAddonDocsIntegration: true,
}
```

..now it's expecting..

```js
'@storybook/ember-cli-storybook': {
  enableAddonDocsIntegration: true,
}
```

😅 

See also, https://github.com/storybookjs/storybook/blob/next/addons/docs/ember/README.md#props-tables

I can fix this issue in the example ember app, or I'm happy to take suggestions on how we might move forwards

**Comment by shilman** [comment]
@BarryThePenguin @storybookjs/ember whatever y'all think is best

**Review by dbendaou — APPROVED** [review]
LGTM! Thanks @BarryThePenguin

**Comment by gossi** [comment]
That change will break for all that have their config under `ember-cli-storybook` key - now SB won't run as expected any longer. I'm fine with the new key here. For the time being we should still look for the old key and throw a deprecation warning, if found under the old key. Question is until when?

**Comment by BarryThePenguin** [comment]
> That change will break for all that have their config under `ember-cli-storybook` key - now SB won't run as expected any longer. I'm fine with the new key here. For the time being we should still look for the old key and throw a deprecation warning, if found under the old key. Question is until when?

Ok, I've created an issue https://github.com/storybookjs/ember-cli-storybook/issues/52

I can put together the relevant PRs too 👍🏻

**Review by gossi — APPROVED** [review]

**Comment by BarryThePenguin** [comment]
So I got around to fixing the issue with the config key.

This branch was quite old, so I rebased my changes onto `next`. Apologies for the force push 😅

File: examples/ember-cli/package.json [hunk 1]
```diff
@@ -28,22 +30,22 @@
...
-    "ember-cli-htmlbars": "^4.4.0",
+    "ember-cli-htmlbars": "^5.6.5",
     "ember-cli-inject-live-reload": "^2.0.2",
     "ember-cli-shims": "^1.2.0",
     "ember-cli-sri": "^2.1.1",
     "ember-cli-uglify": "^3.0.0",
     "ember-load-initializers": "^2.1.2",
-    "ember-resolver": "^7.0.0",
+    "ember-resolver": "^8.0.2",
     "ember-source": "~3.24.0",
```
Comment (gossi) [inline examples/ember-cli/package.json:49]: already wanna go up to `3.25`?

File: examples/ember-cli/package.json [hunk 2]
```diff
@@ -17,6 +17,8 @@
   "devDependencies": {
     "@babel/core": "^7.12.10",
     "@ember/optional-features": "^2.0.0",
+    "@glimmer/component": "^1.0.2",
+    "@glimmer/tracking": "^1.0.2",
```
Comment (gossi) [inline examples/ember-cli/package.json:21]: what about `^1.0.4` ?

**Pull request closed without merging**